astroid/nodes/node_classes.py

```diff
@@ -386,8 +386,12 @@ def ilookup(self, name):
         context = InferenceContext()
         return _infer_stmts(stmts, context, frame)
 
-    def _get_filtered_node_statements(self, nodes):
-        statements = [(node, node.statement()) for node in nodes]
+    def _get_filtered_node_statements(
+        self, nodes: typing.List[NodeNG]
+    ) -> typing.List[typing.Tuple[NodeNG, Statement]]:
+        statements: typing.List[typing.Tuple[NodeNG, Statement]] = [
```

Comment:
```suggestion
        statements = [
```
The type can be inferred from the list comprehension. It's only really necessary for empty lists, if the type can't be inferred, or if you would like to accept a parent class.

```py
l: list[Base] = [childA]
l.append(childB)
```
Usually `l` would have been inferred as `list[ChildA]` and thus the `append` call would have failed.